python/plugins/processing/ProcessingPlugin.py

```diff
@@ -282,6 +283,57 @@ def initGui(self):
         self.iface.actionToggleEditing().triggered.connect(partial(self.sync_in_place_button_state, None))
         self.sync_in_place_button_state()
 
+        self.projectProvider = QgsApplication.instance().processingRegistry().providerById("project")
+        self.projectProvider.algorithmsLoaded.connect(self.updateProjectModelMenu)
+        self.projectModelsMenu = None
+        self.projectMenuAction = None
+        self.projectMenuSeparator = None
+
+    def executeProjectModel(self, model, as_batch=False):
+        """Executes a project model"""
+
+        m = self.toolbox.algorithmTree.model()
+
+        # Search for model
```

Comment:
This looks like quite a fragile approach -- why do we need to go through the toolbox to do this?